UI_Engine/Compute/LoadNewAssemblies.cs

```diff
@@ -0,0 +1,82 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2025, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using BH.oM.UI;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace BH.Engine.UI
+{
+    public static partial class Compute
+    {
+        /*************************************/
+        /**** Public Methods              ****/
+        /*************************************/
+
+        [Description("Loads all BHoM assemblies from the current domain.")]
+        [Input("lastAssemblyUpdateTimes", "records of the last time each assembly was updated.")]
+        [Output("loadedAssemblies", "Assemblies loaded as considered new")]
+        public static List<string> LoadNewAssemblies(Dictionary<string, DateTime> lastAssemblyUpdateTimes)
+        {
+            List<string> loadedAssemblies = new List<string>();
+
+            Regex regex = new Regex(@"oM$|_Engine$|_Adapter$");
+            foreach (string file in Directory.GetFiles(BH.Engine.Base.Query.BHoMFolder(), "*.dll", SearchOption.TopDirectoryOnly))
+            {
+                string name = Path.GetFileNameWithoutExtension(file);
+                if (regex.IsMatch(name))
```

Comment:
Following my previous comment, I would suggest using the existing methods to centralise the core classification of assemblies
```suggestion
                if (name.IsOmAssembly() || name.IsEngineAssembly() || name.IsAdapterAssembly())
```